File: Crio/crio_programming_interview_problems-master/DeleteMiddleNode/DeleteMiddleNode.py
from Solution import *
from crio.ds.List.ListNode import *
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def print_list(head):
    while head != None:
        print(head.val, end = " ")
        head = head.next

def get_nth_node(head, n):
    for i in range(n):
        head = head.next

    return head 

def main():
   n = int(input().strip())
   lst = input().strip().split()
   lst = [int(x) for x in lst]
   k = int(input().strip()) 
   lst = createList(lst)
   assert(k > 1 and k < n)
   
   res_lst = delete_middle_node(lst, get_nth_node(lst, k-1))
   print_list(res_lst)

if __name__ == '__main__':
    main()

File: Crio/crio_programming_interview_problems-master/DeleteMiddleNode/DeleteMiddleNode.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

ListNode* getKthNode(ListNode* head, int k) {
    if(k == 1) {
        return head;
    }
    return getKthNode(head->next , k - 1);
}

int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
    }
    int k;
    cin >> k;

    ListNode* head = ListNode().createList(list);
    ListNode* nodeToDelete = getKthNode(head , k);
    deleteMiddleNode(nodeToDelete);
    vector<int> resultList = ListNode().extractList(head);
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}
